urlc8dbc7ce6e_dlubarov_guava_tgz-pJ8-MainJ8/src/parse/stm/SwitchParser.java

```diff
@@ -0,0 +1,161 @@
+package parse.stm;
+
+import java.util.*;
+
+import common.NiftyException;
+
+import parse.*;
+import parse.exp.ExpressionParser;
+import parse.misc.*;
+import util.StringUtils;
+import a.Type;
+import a.exp.*;
+import a.stm.*;
+import a.stm.Switch.*;
+
+public class SwitchParser extends Parser<Statement> {
+    public static final Parser<Statement> singleton = new SwitchParser();
+    private SwitchParser() {}
+
+    @Override
+    public Success<Statement> parse(String s, int p) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 54-87 into a separate method.